src/lsps1/event.rs

```diff
@@ -10,64 +19,104 @@ use bitcoin::secp256k1::PublicKey;
-		/// TODO
+		/// The website of the LSP.
 		website: String,
-		/// TODO
+		/// All options supported by the LSP.
 		options_supported: OptionsSupported,
 	},
-	/// TODO
+	/// Confirmation from the LSP about the order created by the client.
+	///
+	/// When the invoice is paid, the LSP will open a channel to you
+	/// with the below agreed upon parameters.
+	///
+	/// You must pay the invoice if you want to continue and then
+	/// call [`LSPS1ClientHandler::check_order_status`] with the order id
+	/// to get information from LSP about progress of the order.
+	///
+	/// [`LSPS1ClientHandler::check_order_status`]: crate::lsps1::client::LSPS1ClientHandler::check_order_status
 	DisplayOrder {
-		/// TODO
+		/// This is a randomly generated identifier used to track the channel state.
+		/// It is not related in anyway to the eventual lightning channel id.
+		/// It needs to be passed to [`LSPS1ClientHandler::check_order_status`].
+		///
+		/// [`LSPS1ClientHandler::check_order_status`]: crate::lsps1::client::LSPS1ClientHandler::check_order_status
 		id: u128,
-		/// TODO
+		/// The node id of the LSP.
 		counterparty_node_id: PublicKey,
-		/// TODO
+		/// The order created by client and approved by LSP.
 		order: OrderParams,
-		/// TODO
+		/// The details regarding payment of the order
 		payment: OrderPayment,
-		/// TODO
+		/// The details regarding state of the channel ordered.
 		channel: Option<ChannelInfo>,
 	},
 }
 
 /// An event which an LSPS1 server should take some action in response to.
 #[derive(Clone, Debug, PartialEq, Eq)]
 pub enum LSPS1ServiceEvent {
-	/// TODO
+	/// A client has selected the parameters to use from the supported options of LSP
+	/// and would like to open a channel with the given invoice.
+	///
+	/// You must call [`LSPS1ServiceHandler::send_invoice_for_order`] to
+	/// generate a complete invoice including the details regarding the
+	/// payment and order id for this order for the client.
+	///
+	/// [`LSPS1ServiceHandler::send_invoice_for_order`]: crate::lsps1::service::LSPS1ServiceHandler::send_invoice_for_order
 	CreateInvoice {
-		/// TODO
+		/// An identifier that must be passed to [`LSPS1ServiceHandler::send_invoice_for_order`].
+		///
+		/// [`LSPS1ServiceHandler::send_invoice_for_order`]: crate::lsps1::service::LSPS1ServiceHandler::send_invoice_for_order
 		request_id: RequestId,
-		/// TODO
+		/// The node id of the client making the information request.
 		counterparty_node_id: PublicKey,
-		/// TODO
+		/// The order requested by the client.
 		order: OrderParams,
 	},
-	/// TODO
+	/// A request from client to check the status of the payment.
+	///
+	/// An event to poll for checking payment status either onchain or lightning.
+	///
+	/// You must call [`LSPS1ServiceHandler::update_order_status`] to update the client
+	/// regarding the status of the payment and order.
+	///
+	/// [`LSPS1ServiceHandler::send_invoice_for_order`]: crate::lsps1::service::LSPS1ServiceHandler::send_invoice_for_order
```

Comment:
This needs to link to the right method:
```suggestion
	/// [`LSPS1ServiceHandler::update_order_status`]: crate::lsps1::service::LSPS1ServiceHandler::update_order_status
```